workflows/methylation/methylation-cohort.wdl

```diff
@@ -0,0 +1,361 @@
+        }
+    }
+
+    call filter_probes { input:
+        beta_values = select_first(
+            [
+                final_merge.combined_beta,
+                simple_merge.combined_beta,
+            ]
+        ),
+    }
+
+    call generate_umap { input:
+        filtered_beta_values = filter_probes.filtered_beta_values,
+    }
+
+    call plot_umap { input:
+        umap = generate_umap.umap,
+    }
+
+    output {
+        File combined_beta = select_first(
+            [
+                final_merge.combined_beta,
+                simple_merge.combined_beta,
+            ]
+        )
+        File filtered_beta = filter_probes.filtered_beta_values
+        File filtered_probes = filter_probes.filtered_probes
+        File umap_embedding = generate_umap.umap
+        File umap_plot = plot_umap.umap_plot
+    }
+}
+
+task combine_data {
+    meta {
+        description: "Combine data from multiple samples"
+        outputs: {
+            combined_beta: "Combined beta values for all samples"
+        }
+    }
+
+    parameter_meta {
+        unfiltered_normalized_beta: "Array of unfiltered normalized beta values for each sample"
+        combined_file_name: "Name of the combined file"
+        addl_memory_gb: "Additional memory to allocate for the task (in GB)"
+    }
+
+    input {
+        Array[File] unfiltered_normalized_beta
+        String combined_file_name = "combined_beta.csv"
+        Int addl_memory_gb = 0
+    }
+
+    Int memory_gb = ceil(size(unfiltered_normalized_beta, "GiB") * 2) + addl_memory_gb
+    Int disk_size_gb = ceil(size(unfiltered_normalized_beta, "GiB") * 2)
+
+    command <<<
+        echo "Combining data"
+        input_files=""
+        for file in ~{sep(" ", unfiltered_normalized_beta)}
+        do
+            #ln -s ~{sep(" ", unfiltered_normalized_beta)} .
+            ln -s $file .
+            input_files="${input_files} $(basename $file)"
+        done
+
+        cat <<SCRIPT > run.py
+        import sys
+        import argparse
+        import pandas as pd
+
+        def get_args():
+            parser = argparse.ArgumentParser(
+                description="Combine CSV files.")
+            parser.add_argument(
+                "csvs", type=str, nargs="+", help="List of CSV files.")
+
+            args = parser.parse_args()
+
+            return args
+        
+        def read(filename):
+            df = pd.read_csv(filename)
+            df.rename(columns={df.columns[0]: "probe"}, inplace=True)
+            df.set_index("probe", inplace=True)
+            return df
+
+        if __name__ == "__main__":
+            args = get_args()
+
+            # Combine data
+            df = pd.concat([read(f) for f in args.csvs], axis=1, join="inner")
+            df.to_csv("~{combined_file_name}", index=True)
+        SCRIPT
+
+        python run.py $input_files
+
+    >>>
+
+    output {
+        File combined_beta = combined_file_name
+    }
+
+    #@ except: ContainerValue
```

Comment:
IDK if we want this here. In the CI we've specified `--except ContainerValue` so this is redundant in that context. I guess the highlight is annoying in the IDE which could be reason enough. But I'm worried about missing this if/when we finally decide to adhere to `ContainerValue`.

Reply:
I was working off an older version of the CI when I resolved the lint warnings. I can remove these now that my branch is up-to-date.